com.ibm.streamsx.topology/com.ibm.streamsx.topology.functional.python/CCWindow/CCWindow_cpp.cgt

```diff
@@ -103,22 +108,31 @@ print splpy_inputtuple2value($pystyle);
 @include "../pyspltuple2dict.cgt"
 <%}%>
 
-  unsigned char const *data = value.getData();
-  unsigned char fmt = *data;
-
   PyObject *python_value;
 
-  if (fmt == STREAMSX_TPP_PTR) {
-      __SPLTuplePyPtr *stp = (__SPLTuplePyPtr *)(data);
-      python_value = stp->pyptr;
-  }
-  // Anything ASCII is also Pickle (Python 2 default format)
-  else if (fmt <= STREAMSX_TPP_PICKLE) {
-      python_value = pySplValueToPyObject(value);
-  }
+// If the input style is pickle,
+{
+  // None of the streamsx::topology methods in this scope grab the lock
+  // so we need to do it here.
+  SplpyGIL lock; 
+  <%if ($pystyle eq 'pickle'){%>
+
+      unsigned char const *data = value.getData();
+      unsigned char fmt = *data;
+      if (fmt == STREAMSX_TPP_PTR) {
+          __SPLTuplePyPtr *stp = (__SPLTuplePyPtr *)(data);
+          python_value = stp->pyptr;
+      }
+      // Anything ASCII is also Pickle (Python 2 default format)
+      else if (fmt <= STREAMSX_TPP_PICKLE) {
+          python_value = pySplValueToPyObject(value);
+      }
+  <%} else {%>
+      python_value = streamsx::topology::Splpy::pyTupleMap(spl_in_object_out->callable(), value);

```

Comment:
I'm still unsure about using `pyTupleMap` here just because is seems to (almost) do the right thing.

Here' we are not *mapping* an object from one value to another, but just preserving the tuple in the window.

Reply:
pyTypleMap has been replaced with 'loadFunction()' and 'pyCallObject()